recOrder/tests/cli_tests/test_cli.py

```diff
@@ -1,10 +1,154 @@
 from recOrder.cli.main import cli
+import unittest
 from click.testing import CliRunner
-
+from unittest.mock import patch, Mock
+from iohub.ngff import open_ome_zarr, Position
+from hypothesis import given, HealthCheck, settings
+from hypothesis import strategies as st
+from recOrder.cli.compute_transfer_function import (
+    generate_and_save_phase_transfer_function, 
+    generate_and_save_birefringence_transfer_function,
+    compute_transfer_function_cli
+)
+from recOrder.cli.settings import (
+    TransferFunctionSettings,
+    _PhaseTransferFunctionSettings,
+    _BirefringenceTransferFunctionSettings,
+    _UniversalSettings
+)
+import pytest
 
 def test_main():
     runner = CliRunner()
     result = runner.invoke(cli)
 
     assert result.exit_code == 0
     assert "Toolkit" in result.output
+
+def test_compute_transfer(tmp_path):
+    runner = CliRunner()
+    result = runner.invoke(cli, "compute-transfer-function -o " + str(tmp_path))
+    assert result.exit_code == 0
+    assert "Generating" in result.output
+
+def test_compute_transfer_blank_config():
+    runner = CliRunner()
+    for option in ("-c ", "--config-path "):
+        cmd = "compute-transfer-function " + option
+        result = runner.invoke(cli, cmd)
+        assert result.exit_code == 2
+        assert "Error" in result.output
+
+def test_compute_transfer_blank_output():
+    runner = CliRunner()
+    for option in ("-o ", "--output-path "):
+        cmd = "compute-transfer-function " + option
+        result = runner.invoke(cli, cmd)
+        assert result.exit_code == 2
+        assert "Error" in result.output
+
+def test_compute_transfer_output_file(tmp_path):
+    runner = CliRunner()
+    paths = ["test1", "test2/test"]
+    for option in ("-o ", "--output-path "):
+        temp_cmd = "compute-transfer-function " + option
+        for path in paths:
+            cmd = temp_cmd + str(tmp_path.joinpath(path))
+            result = runner.invoke(cli, cmd)
+            assert result.exit_code == 0
+            assert path in result.output
+            assert tmp_path.exists()
+
+def test_compute_transfer_config_none(tmp_path):
+    runner = CliRunner()
+    for option in ("-c ", "--config-path "):
+        cmd = "compute-transfer-function " + option + "None " + "-o " + str(tmp_path)
+        result = runner.invoke(cli, cmd)
+        assert result.exit_code == 0
+        assert "Generating" in result.output
+
+def test_stokes_matrix_write(setup_default_ctf_settings):
+    settings, dataset = setup_default_ctf_settings
+    generate_and_save_birefringence_transfer_function(settings, dataset)
+    assert dataset["intensity_to_stokes_matrix"]
+
+def test_absorption_and_phase_write(setup_default_ctf_settings):
+    settings, dataset = setup_default_ctf_settings
+    generate_and_save_phase_transfer_function(settings, dataset)
+    assert dataset["absorption_transfer_function"]
+    assert dataset["phase_transfer_function"]
+    with pytest.raises(KeyError):
```

Comment:
This is not testing for an error raised from the tested function. To test that the key does not exist in a dict in the local scope simply do:

```py
assert "key1" not in dataset
assert "key2" not in dataset
```

which is much faster.